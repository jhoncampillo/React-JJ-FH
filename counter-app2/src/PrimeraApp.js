import React from "react";
import { Fragment } from "react/cjs/react.production.min";
//para importar prop-types
import PropTypes from "prop-types";
//imr + tab
//Funcional Component
//RECIBO LAS PROPS
const PrimeraApp = ({ saludo, subtitulo }) => {
  //console.log(props);
  //const saludo = "Hola Mundo Jhon2 React";
  //const saludo2 = [1, 2, 3, 4, 5, 6, 7];

  //Manejo de objetos
  const usuario = {
    nombre: "Jhon Jairo Campillo Serrato",
    Edad: 49,
    Profesion: "Ing  de sistemas",
  };

  //retornmo objeto al envolverlo en parentesis
  //otro fragmente : <> </>  o  <div></div> o   <Fragment>  </Fragment>
  return (
    <>
      {/* Puede pasarse como una props.saludos3 pero se usa destrusturing */}
      {/* <h1>{saludo3} </h1> */}
      {/* <pre>{JSON.stringify(usuario, null, 3)}</pre> */}
      {/* <h2>{saludo2} </h2> */}
      <h2>{saludo} </h2>
      {/* <h1>Hola Mundo React de Jhon</h1>
      <p>Primera Aplicaicon</p> */}
      <p>{subtitulo}</p>
    </>
  );
};
//defino el typo de las propiedades de este componente
PrimeraApp.propTypes = {
  //es requerido el saludo en el componente.
  saludo: PropTypes.string.isRequired,
};
PrimeraApp.defaultProps = {
  subtitulo: "Soy un Subtitulo",
};

//exporto
export default PrimeraApp;
