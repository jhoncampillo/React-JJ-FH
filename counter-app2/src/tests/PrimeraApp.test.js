//import { render } from "@testing-library/react";
import React from "react";
//import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import PrimeraApp from "../PrimeraApp";

describe("Pruebas en PrimeraApp", () => {
  //   test('Debe mostrara el mensaje "Hola soy JHon Jairo Devops" ', () => {
  //     const saludo = "Hola soy JHon Jairo Devops";
  //     //tengo que enviar la props saludo por que es requeried
  //     //el wrapper tiene mucho metodos entonses desesctructuro y uso el getByText

  //     // const wrapper = render(<PrimeraApp saludo={saludo} />);
  //     const { getByText } = render(<PrimeraApp saludo={saludo} />);
  //     //evaluo si el porop saludo esta en el componente
  //     expect(getByText(saludo)).toBeInTheDocument();
  //   });

  //CAMBIO A PRUEBAS CON ENZYME
  test("Debe Mostrar  <PrimeraApp /> correctamente", () => {
    //Creo el saludo
    const saludo = "Hola soy JHon Jairo Devops";

    //shallow parecido al renden
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    //espero que se renderice
    //toMatchSnapshot() =  crea una imagen exacta del html como esta antes de la prueba

    expect(wrapper).toMatchSnapshot();
  });
  //Prueb apar el subtitulo
  test("Debe mostrar el  subtitulos enviado por el props ", () => {
    const saludo = "Hola soy JHon Jairo Devops";
    const subTitulo = "Soy un subtitulo";

    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subTitulo} />
    );

    //wrapper tiene toda la informacion del componente renderizado
    //el find trabaja igual; que DQS - busca el parrafo como el document,queryselector-false . - # id
    const textoParrafo = wrapper.find("p").text();
    console.log(textoParrafo);
    expect(textoParrafo).toBe(subTitulo);
  });
});

//Se instala para poder tener un snpshot renderizado de la prueba
//npm install --save-dev enzyme-to-json
