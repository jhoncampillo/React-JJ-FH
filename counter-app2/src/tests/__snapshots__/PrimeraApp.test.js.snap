// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp Debe Mostrar  <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h2>
    Hola soy JHon Jairo Devops
     
  </h2>
  <p>
    Soy un Subtitulo
  </p>
</Fragment>
`;
